operator/pkg/controller/virtualippool/controller.go

```diff
@@ -57,7 +75,177 @@ func (r *VirtualIPPoolReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return utils.Requeue(err)
 	}
 
+	// Retrieve all nodes
+	if err := r.client.List(ctx, &r.nodeList); err != nil {
+		return utils.Requeue(err)
+	}
+
+	// Retrieve all configured Virtual Router IDs
+	r.usedVRID = make(map[int]bool, 255)
+	pools := metalk8sscalitycomv1alpha1.VirtualIPPoolList{}
+	if err := r.client.List(ctx, &pools); err != nil {
+		return utils.Requeue(err)
+	}
+	for _, pool := range pools.Items {
+		if err := r.retrieveVRID(ctx, pool); err != nil {
+			return utils.Requeue(err)
+		}
+	}
```

Comment:
I agree that it's not ideal but I don't see any better approach.

> Still, it would be cleaner (IMO) to have each Pool reconciliation update the cache (about the VRIDs it's using) without having to recreate it on each pass for each Pool.

It will not work well, it means the first VirtualIPPool to get reconciled may take one VRID already taken by another Pool, or we need to load all the pools at the startup of the controller but not ideal as well.

TBH, we will never have tonnes of VirtualIPPools at once so I don't think it's really a problem, even if I agree it's not ideal.